src/main/java/com/scaleunlimited/flinkcrawler/urls/SimpleUrlLengthener.java

```diff
@@ -1,21 +1,118 @@
 package com.scaleunlimited.flinkcrawler.urls;
 
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.http.HttpStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.scaleunlimited.flinkcrawler.fetcher.BaseHttpFetcherBuilder;
 import com.scaleunlimited.flinkcrawler.pojos.RawUrl;
 
+import crawlercommons.fetcher.BaseFetchException;
+import crawlercommons.fetcher.FetchedResult;
+import crawlercommons.fetcher.RedirectFetchException;
+import crawlercommons.fetcher.http.BaseHttpFetcher;
+
 @SuppressWarnings("serial")
 public class SimpleUrlLengthener extends BaseUrlLengthener {
+    private static final Logger LOGGER = LoggerFactory.getLogger(SimpleUrlLengthener.class);
 
-	@Override
+    private static final Pattern HOSTNAME_PATTERN = Pattern.compile("^http://([^/:?]{3,})");
+
+    private BaseHttpFetcher _fetcher;
+    private Set<String> _urlShorteners;
+
+    public SimpleUrlLengthener(BaseHttpFetcherBuilder fetcherBuilder) {
```

Comment:
I think we need an `open()` method in `BaseUrlLengthener`, and we only build the fetcher in that open call. And that needs to be called in the `open()` method of the `LengthenUrlsFunction`. Otherwise we're building the fetcher when we're building the topology, which is too soon.

Reply:
Done